### PR TITLE
fix: Fix 30 mins diff tests show hundreds of thousands of errors

### LINKED ISSUE #11709: Fix 30 mins diff tests show hundreds of thousands of errors
### Description

Round 162900035, 30 minute diff test came up with 699,230 differences.

### Steps to reproduce

run 30 mins diff test on this round.

### Additional context

_No response_

### Hedera network

other

### Version

develop

### Operating system

None

### PR DESCRIPTION
**Pull request by iwsimon**
**Description**:

Round 162900035, 30 minute diff test dropped from 699,230 to 181. Logged more details if no modular records found or additional modular records found comparing with mono records in diffs.txt.

**Related issue(s)**:

Fixes #11709

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by povolev15 — APPROVED** [review]

**Review by tinker-michaelj — APPROVED** [review]
Ok, _this_ looks obviously correct 😁 Tyvm @iwsimon !

**Review by kimbor — APPROVED** [review]

File: hedera-node/hapi-utils/src/main/java/com/hedera/node/app/hapi/utils/forensics/OrderedComparison.java [hunk 1]
```diff
@@ -68,40 +68,68 @@ public static List<DifferingEntries> findDifferencesBetweenV6(
...
-                secondIndex++;
-            } else {
-                ret.add(new RecordStreamEntry(null, null, firstEntry.consensusTime()));
+        List<RecordStreamEntry> firstList = new ArrayList<>();
+        List<RecordStreamEntry> secondList = new ArrayList<>();
+
+        if (secondEntries.isEmpty() || firstEntries.isEmpty()) {
+            return new CompareList(firstEntries, secondEntries);
+        } else {
+            int i = 0, j = 0;
```
Comment (mhess-swl) [inline hedera-node/hapi-utils/src/main/java/com/hedera/node/app/hapi/utils/forensics/OrderedComparison.java:88]: Nit: codacy is going to start flagging these soon, based on the PRs I've seen from Nathan Klick
```
int i = 0;
int j = 0;
```

Comment (iwsimon) [reply hedera-node/hapi-utils/src/main/java/com/hedera/node/app/hapi/utils/forensics/OrderedComparison.java]: I don't completely agree with codacy. Fixed it here.

**Comment by mhess-swl** [comment]
lol me either 😆

Best,

*Matt Hess*

Senior Software Engineer

*SWIRLDS LABS*

E: ***@***.***

W: www.swirldslabs.com <http://www.swilrdslabs.com>

**Review by povolev15 — APPROVED** [review]

**Review by mhess-swl — APPROVED** [review]

**Review by kimbor — APPROVED** [review]

**Pull request merged**